### PR TITLE
Alert Email Template Refine

### PR DESCRIPTION
**Pull request by suiguoxin**
* add kill user job alert template #5021
* add troubleshooting link in general email template
* redefine `actions` schema in `customized-receivers` to facilitate parameters passing

Email Examples:
User:
image

Admin:
image

### CODE REVIEW

File: src/alert-manager/src/alert-handler/routes/actions.js [hunk 1]
```diff
@@ -24,12 +24,12 @@ const router = express.Router();
 
 // email
 router
-  .route('/alert-handler/send-email-to-admin')
+  .route('/alert-handler/send-email-to-admin/:template*?')
```
Comment (Binyang2014) [inline src/alert-manager/src/alert-handler/routes/actions.js:27]: Change API to `/alert-handler/send-email-to-admin?template=template-name`?

**Comment by Binyang2014** [comment]
Just wondering if we can put all `ejs` into separate folder, such as template folder and not build them into docker image (We can make them as a configmap and mount into container). So if customer want to change/add some templates, they just need to change some config.

File: contrib/kubespray/quick-start/services-configuration.yaml.template [hunk 2]
```diff
@@ -183,16 +183,18 @@ authentication:
...
-#     - email-admin
-#     - email-user
-#     - stop-jobs
-#     - tag-jobs
-#     tags: 
-#     - 'stopped-by-alert-manager'
-
+#       # the email template for `email-admin` and `email-user `can be chosen from ['general-template', 'kill-low-efficiency-job-alert']
+#       # if no template specified, 'general-template' will be used.
+#       email-admin: No-Parameters-Required
```
[the remark this answers is not in the document]
Comment (suiguoxin) [reply contrib/kubespray/quick-start/services-configuration.yaml.template:191]: > Change to:
> 
> ```
> #       email-admin:  # No-Parameters-Required
> ```
> 
> So only need to delete first `#`

`No-Parameters-Required` is the value of key `email-admin`. Here we change the  value of `Actions` from a list to a dict. Simply `email-admin: ` doesn't conform to the yaml format. Users can assign whatever as the value if they don't need to specify the template.

[the remark this answers is not in the document]
Comment (Binyang2014) [reply contrib/kubespray/quick-start/services-configuration.yaml.template:191]: Can this work?
```yaml
 email-admin: {}
```

File: src/alert-manager/src/alert-handler/routes/actions.js (hunk 1 above)
Comment (suiguoxin) [reply src/alert-manager/src/alert-handler/routes/actions.js]: > Change API to `/alert-handler/send-email-to-admin?template=template-name`?

Changed, thanks.

File: contrib/kubespray/quick-start/services-configuration.yaml.template (hunk 2 above)
[the remark this answers is not in the document]
Comment (suiguoxin) [reply contrib/kubespray/quick-start/services-configuration.yaml.template]: > Can this work?
> 
> ```yaml
>  email-admin: {}
> ```

Modified, thanks

File: contrib/kubespray/quick-start/services-configuration.yaml.template [hunk 3]
```diff
@@ -178,21 +178,25 @@ authentication:
 #       smtp-from: alert-sender@example.com
 #       smtp-auth-username: alert-sender@example.com
 #       smtp-auth-password: password-for-alert-sender
+#       customized-templates: # optional
+#       - your-customized-email-template
```
Comment (Binyang2014) [inline contrib/kubespray/quick-start/services-configuration.yaml.template:182]: Do we really need this option? Can we parse the folder to get the templates? I think you can modify the deploy python file to achieve this

Comment (suiguoxin) [reply contrib/kubespray/quick-start/services-configuration.yaml.template:182]: > Do we really need this option? Can we parse the folder to get the templates? I think you can modify the deploy python file to achieve this

Right, but I don't want to touch the python2 code...

Comment (suiguoxin) [reply contrib/kubespray/quick-start/services-configuration.yaml.template:182]: > Do we really need this option? Can we parse the folder to get the templates? I think you can modify the deploy python file to achieve this

How can we save the parsing results in `cluster_cfg` ?

Comment (Binyang2014) [reply contrib/kubespray/quick-start/services-configuration.yaml.template:182]: You can change https://github.com/microsoft/pai/blob/df25b985699b75a3813a2080ee6d437ca0db1bf2/src/alert-manager/config/alert_manager.py#L66-L69. Put the parsed result into `result` and return. Then you will get it from `cluster_cfg`

Comment (suiguoxin) [reply contrib/kubespray/quick-start/services-configuration.yaml.template:182]: > You can change
> 
> https://github.com/microsoft/pai/blob/df25b985699b75a3813a2080ee6d437ca0db1bf2/src/alert-manager/config/alert_manager.py#L66-L69
> 
> . Put the parsed result into `result` and return. Then you will get it from `cluster_cfg`

`alert_manager` is a class definition file, there is no guarantee on the path to pass

File: src/alert-manager/deploy/start.sh.template [hunk 4]
```diff
@@ -19,6 +19,18 @@
...
+{% if 'email-admin' in cluster_cfg["alert-manager"]["actions-available"] %}
+kubectl create configmap alert-templates \
+{% for template in cluster_cfg["alert-manager"]["alert-handler"]["email-configs"]["templates"] -%}
+--from-file={{ template }}-html.ejs=alert-templates/{{ template }}/html.ejs \
+--from-file={{ template }}-subject.ejs=alert-templates/{{ template }}/subject.ejs \
+{% endfor -%}
+--dry-run -o yaml | kubectl apply --overwrite=true -f - || exit $?
+{% endif %}
+{% endif %}
+
```
Comment (Binyang2014) [inline src/alert-manager/deploy/start.sh.template:33]: Please delete this configmap in stop.sh

**Review by Binyang2014 — APPROVED** [review]

File: src/alert-manager/deploy/start.sh.template (hunk 4 above)
Comment (suiguoxin) [reply src/alert-manager/deploy/start.sh.template]: > Please delete this configmap in stop.sh

Added, thanks

**Pull request merged**